dandi/dandiapi.py

```diff
@@ -699,13 +802,45 @@ def api_path(self) -> str:
     def download_url(self) -> str:
         return self.client.get_url(f"{self.api_path}download/")
 
+    def get_metadata(self) -> models.Asset:
+        """
+        Fetch the metadata for the asset as a `dandischema.models.Asset`
+        instance
+        """
+        return models.Asset.parse_obj(self.get_raw_metadata())
+
     def get_raw_metadata(self) -> Dict[str, Any]:
         """Fetch the metadata for the asset as an unprocessed `dict`"""
         if self._metadata is not None:
             return self._metadata
         else:
             return cast(Dict[str, Any], self.client.get(self.api_path))
 
+    def get_digest(self, digest_type: Union[str, models.DigestType]) -> str:
+        """
+        Retrieves the value of the given type of digest from the asset's
+        metadata.  Raises `NotFoundError` if there is no entry for the given
+        digest type.
+        """
+        if isinstance(digest_type, models.DigestType):
+            digest_type = digest_type.value
+        metadata = self.get_raw_metadata()
+        try:
+            return metadata["digest"][digest_type]
+        except KeyError:
+            raise NotFoundError(f"No {digest_type} digest found in metadata")
+
+    def set_metadata(self, metadata: models.Asset) -> None:
+        """Set the metadata for the asset to the given value"""
+        self.set_raw_metadata(metadata.json_dict())
+
+    def set_raw_metadata(self, metadata: Dict[str, Any]) -> None:
+        """Set the metadata for the asset to the given value"""
+        self.client.put(
+            self.api_path, json={"metadata": metadata, "blob_id": self.identifier}
```

Comment:
hm, why asset's id is provided as `blob_id` here? isn't it an asset identifier?

Reply:
note: discussion is ongoing with @dchiquito in slack internal about this blob_id etc